Helper/Search/TextValue.php

```diff
@@ -6,63 +6,63 @@
  * If we search for 'foo bar'
  * the SearchManager will create two SearchValue instances
  */
-class SearchValue
+class TextValue
 {
     /** @var string */
-    private $term;
+    private $text;
+    /** @var string */
+    private $field;
+    /** @var string */
+    private $analyzer;
+
     /** @var array */
     private $synonyms;
 
-    public function __construct(string $term)
+    public function __construct(string $text, string $field, string $analyzer)
     {
-        $this->term = $term;
+        $this->text = $text;
+        $this->field = $field;
+        $this->analyzer = $analyzer;
         $this->synonyms = [];
     }
 
-    public function addSynonym(array $document): void
+    public function getAnalyzer(): string
     {
-        $this->synonyms[] = sprintf('%s:%s', $document['_type'], $document['_id']);
+        return $this->analyzer;
     }
 
-    public function getTerm(): string
+    public function addSynonym(string $synonymField, array $doc): void
     {
-        return $this->term;
+        $this->synonyms[] = [
+            'match' => [
+                $synonymField => [
+                    'query' => sprintf('%s:%s', $doc['_source']['_contenttype'], $doc['_id']),
```

Comment:
won't this break backward compatibility?

Reply:
Every es document should have the field _contenttype, otherwise we should rebuild the indexes!.